src/org/geekhub/json/JsonSerializer.java

```diff
@@ -70,7 +74,28 @@ public static Object serialize(Object o) {
      * @throws InstantiationException
      */
     private static JSONObject toJsonObject(Object o) throws Exception {
-        //implement me
-        return null;
+        JSONObject jsonObject = new JSONObject();
+
+        Class objectClass = o.getClass();
+        Field[] fields = objectClass.getDeclaredFields();
+        for (Field field : fields) {
+            field.setAccessible(true);
+            if (field.isAnnotationPresent(Ignore.class)) {
+                continue;
+            }
+            if (field.isAnnotationPresent(UseDataAdapter.class)) {
+                Annotation annotation = field.getAnnotation(UseDataAdapter.class);
+                Class<JsonDataAdapter> adapter = (Class<JsonDataAdapter>) ((UseDataAdapter) annotation).value();
```

Comment:
You can use something like this.
JsonDataAdapter dataAdapter = field.getAnnotation(UseDataAdapter.class).value().newInstance();
In this case you can call toJson method directly.